Move session[:userrole] out of session

kbrock opened the pull request:
part of #3519 - no ui change

lazily access `userrole` based upon `current_group`.

Also replace string comparisons with `admin_user?` / `super_admin_user?`

please note: (a || !a && b) === (a || b)

These will be replaced with `current_user.admin_user?` as soon as we are sure that session's `current_group` is the same as `current_user.current_group`.

/cc @dclarizio @gmcculloug

Fryguy commented on app/controllers/application_controller/current_user.rb:
```diff
@@ -42,6 +38,26 @@ def eligible_groups
+      begin
+        role = current_group.try(:miq_user_role)
+        role.try(:read_only?) ? role.name.split("-").last : ""
+      end
```
style only, but most other `||= being; end` pattern looks like this:

```ruby
    @current_role ||= begin
      role = current_group.try(:miq_user_role)
      role.try(:read_only?) ? role.name.split("-").last : ""
    end
```

Fryguy commented on app/services/privilege_checker_service.rb:
```diff
@@ -22,6 +22,6 @@ def session_active?(session)
 
   def server_ready?(session)
-    MiqServer.my_server(true).logon_status == :ready || session[:userrole] == "super_administrator"
+    MiqServer.my_server(true).logon_status == :ready || super_admin_user?
```
Interesting...if you flip this logic, then it hits the database less frequently in super admin cases...it's funny how refactoring makes things like this pop up.

Fryguy commented on app/views/layouts/_adv_search_footer.html.haml:
```diff
@@ -34,9 +34,7 @@
-      - if (session[:userrole] == "super_administrator" || session[:userrole] == "administrator") ||
-      -    (session[:userrole] != "super_administrator" && session[:userrole] != "administrator" &&
-      -    @edit[@expkey][:selected][:typ] == "user")
+      - if admin_user? || @edit[@expkey][:selected][:typ] == "user")
```
Extra bonus paren on the right?  Shouldn't this be a syntax error.

dclarizio replied:
@kbrock  This causes a bunch of tests to fail with unexpected ) error.

Fryguy commented:
:heart: this change @kbrock !

chessbyte commented:
@Fryguy @kbrock this PR rocks!

dclarizio commented on app/services/user_validation_service.rb:
```diff
@@ -87,7 +87,7 @@ def missing_user_features(db_user)
   end
 
   def user_is_super_admin?
```
@kbrock I'm thinking this can be removed and the couple places that call it replaced with the new helper version.

kbrock replied:
a) This calls into that helper. Since that is not visible, it uses `controller#super_admin_user` to access it.
b) I have the goal of setting `user#current_group` correctly, then we can just use `db_user#super_admin_user?` and dump the helper / call into the controller.

chessbyte commented:
@kbrock Travis is still failing. :-(

kbrock commented:
@chessbyte great question.

Here is what I have uncovered, @dclarizio may have a different view.

Sprint 59 introduced setting the user's role to `db_user.role.name`. It looks like `'super_user'`.

Sprint 69 ( 3/7/2011 )moved over to using `role.id`. The supplied roles (they have the `read_only` flag set) were prefixed with `EvmRole-`.
6a9b80a4b4dce131761fe3e6ac2f5346d62ffd73  

Since there was a bunch of logic that did string comparison on the original value (pre-sprint 69), it was included in the session as well.

This PR centralize that string comparison.

I'll remove the comment

kbrock commented:
man, I've introduced the same syntax error every time I've rebased `app/views/layouts/_adv_search_footer.html.haml` - fixed along with a warning.

My dear rubocop,

- app/helpers/application_helper/toolbar_chooser.rb:73 is too much for me to fix
- `!!` returns a false, I want to keep the return of the methods the same for now. Hope to remove that all together in the near future.
